### PR TITLE
try catch all the promises after reservation its completed

### LINKED ISSUE #5079: [CAL-180] Booking Reschedule not working.
### Issue Summary

After scheduling a booking, I tried to reschedule it. I did not get updated.

### Steps to Reproduce

1. Clicked on someone's link to event.
2. Selected a time and booked a time.
3. On the next page "This meeting is scheduled" clicked Reschedule
4. Selected a different time and submitted.
5. Did not received a new email with the updated link nor the calendar event has changed.

### Another way to reproduce after step 2 is:

1. Went to my email, selected the email with the confirmation. Clicked to manage the booking.
2. Selected to reschedule the meeting.
3. Selected a different time for the meeting and confirmed.
4. Did not receive an updated email confirmation, nor the host of event-type.

### Technical details

I tested on actual [cal.com](http://cal.com)

> From [Linear-GitHub Sync](https://synclinear.com)

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

Add try catch in promises after reservation booking has completed.
NOTE: the main problem that was reported for this case was, an update for scheduled jobs when reschedule its happening code was trying to update scheduled jobs on already deleted booking due to reschedule.

Fixes #5079 

**Environment**: Staging(main branch) / Production

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] Force an exception while canceling scheduled jobs after booking has completed.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 1]
```diff
@@ -882,85 +882,108 @@ async function handler(req: NextApiRequest & { userId?: number | undefined }) {
...
+    log.error("Error while updating hashed link", error);
+  }
+
+  if (!booking) throw new HttpError({ statusCode: 400, message: "Booking failed" });
+
+  try {
+    await prisma.booking.update({
       where: {
-        link: reqBody.hashedLink as string,
+        uid: booking.uid,
```
Comment (alannnc) [inline packages/features/bookings/lib/handleNewBooking.ts:961]: this change belongs to referencesToCreate and not hashedLink

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 2]
```diff
@@ -882,85 +882,108 @@ async function handler(req: NextApiRequest & { userId?: number | undefined }) {
...
+    subscribersMeetingEnded.forEach((subscriber) => {
+      if (rescheduleUid && originalRescheduledBooking) {
+        cancelScheduledJobs(originalRescheduledBooking);
+      }
+      if (booking && booking.status === BookingStatus.ACCEPTED) {
+        scheduleTrigger(booking, subscriber.subscriberUrl, subscriber);
+      }
+    });
+  } catch (error) {
+    log.error("Error while running scheduledJobs for booking", error);
```
Comment (harshsinghatz) [inline packages/features/bookings/lib/handleNewBooking.ts:897]: Just a doubt, shouldn't this be console.error("...")

Comment (alannnc) [reply packages/features/bookings/lib/handleNewBooking.ts:897]: It's an internal logger we use sometimes.
`const log = logger.getChildLogger({ prefix: ["[api] book:user"] });`

**Review by zomars — APPROVED** [review]
Definitively and improvement Thanks @alannnc 

Ship it

**Pull request merged**